### PR TITLE
Make reusable block e2e tests independent.

### LINKED ISSUE #21249: Reusable Blocks: Move E2E tests setup logic to a beforeAll()
The tests in `packages/e2e-tests/specs/editor/various/reusable-blocks.test.js` have an assumed order which is incorrect as tests should be able to run independently of one another.

From https://github.com/WordPress/gutenberg/pull/7947#discussion_r399341588:

> There should ideally be no concept of "above" as assuming tests to be run in particular order (or even that any others are run at all). I should be able to change this to [`it.only`](https://jestjs.io/docs/en/api#testonlyname-fn-timeout) and the test should be fine to run in isolation. Common behaviors should be established in lifecycle (`beforeEach` or `beforeAll`) or repeated in every test case.
>
> Note: I'm encountering this in implementing a new end-to-end test case for verifying the fix at #21205. Ideally I'd like to iterate on the test case with `it.only`, but at least as currently presented, there's no prior art for me to leverage commonly available reusable blocks to work with. For the purposes of my test, I might just copy the logic from "can be created" to create the reusable block. It could be straight-forward enough to refactor to its own function; not sure if I'll do that or end up just following the pattern here of reusing the block created in an earlier test case once finished.

### PR DESCRIPTION
**Pull request by sainthkh**
* Closes #21249

## Description

Made reusable block e2e tests independent.

## How has this been tested?

e2e tested with code.

## Screenshots 

N/A

## Types of changes

Bug fix

## Checklist:
- [x] My code is tested.
- [N/A] My code follows the WordPress code style. 
- [N/A] My code follows the accessibility standards. 
- [N/A] My code has proper inline documentation. 
- [N/A] I've included developer documentation if appropriate. 
- [N/A] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/e2e-tests/specs/editor/various/reusable-blocks.test.js [hunk 1]
```diff
@@ -165,40 +209,19 @@ describe( 'Reusable blocks', () => {
 	} );
 
 	it( 'can be inserted after refresh', async () => {
```
Comment (sainthkh) [inline packages/e2e-tests/specs/editor/various/reusable-blocks.test.js:211]: After investigating, I learned that `can be inserted and edited` and `can be inserted after refresh` are different tests.

* `can be inserted and edited` focuses on testing creating and editing in one writing session.
* `can be inserted after refresh` focuses on testing the reusable block is saved correctly to the database and loaded without fail when writing a new post or editing existing ones.

File: packages/e2e-tests/specs/editor/various/reusable-blocks.test.js [hunk 2]
```diff
@@ -16,94 +17,137 @@ function waitForAndAcceptDialog() {
...
-		await page.evaluate( () => {
-			const blocks = wp.data.select( 'core/block-editor' ).getBlocks();
-			const clientIds = blocks.map( ( block ) => block.clientId );
-			wp.data.dispatch( 'core/block-editor' ).removeBlocks( clientIds );
-		} );
-	} );
+	// When no reusable block is registered, return
+	if ( ( await page.$x( '//td[text()="No blocks found."]' ) ) !== null ) {
+		return;
+	}
```
Comment (sainthkh) [inline packages/e2e-tests/specs/editor/various/reusable-blocks.test.js:26]: It was a bit hard to find that this case was failing the tests.

**Comment by sainthkh** [comment]
@noisysocks It's ready for review. This PR:

* removes reusable blocks if any.
* creates new reusable block for each test.

**Comment by youknowriad** [comment]
Thanks for your work here. For some reason, I missed this PR :( 

I did an extensive refactor to this test a few weeks ago and I believe it's now based on independent tests.

**Pull request closed without merging**